Spring-Boot/Spring-Boot-Problem/src/main/java/tw/waterball/judgegirl/springboot/problem/repositories/MongoProblemRepository.java

```diff
@@ -131,6 +133,27 @@ public int saveProblemWithTitleAndGetId(String title) {
         return id;
     }
 
+    @Override
+    public void patchProblem(int problemId, PatchProblemParams params) {
+        Update update = new Update();
+        Query query = new Query(where("_id").is(problemId));
+        params.getTitle().ifPresent(title -> update.set("title", title));
+        params.getDescription().ifPresent(des -> update.set("description", des));
+        params.getMatchPolicyPluginTag().ifPresent(tag ->
+                update.set("outputMatchPolicyPluginTag", tag));
+        params.getFilterPluginTags().ifPresent(tags -> {
+            if (!tags.isEmpty()) {
+                update.addToSet("filterPluginTags").each(tags);
+            }
```

Comment:
it's fine if it's empty

since the client might want to empty the tags.

and it's not **adding** it's **setting**
we are replacing all the previous tags with the new set of tags.